x-pack/plugins/ml/public/settings/calendar/list/delete_calendars.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { toastNotifications } from 'ui/notify';
+import { ml } from '../../../services/ml_api_service';
+
+
+export async function deleteCalendars(calendarsToDelete, callback) {
+  if (calendarsToDelete === undefined || calendarsToDelete.length === 0) {
+    return;
+  }
+
+  // Delete each of the specified calendars in turn, waiting for each response
+  // before deleting the next to minimize load on the cluster.
+  const messageId = `${(calendarsToDelete.length > 1) ?
+    `${calendarsToDelete.length} calendars` : calendarsToDelete[0].calendar_id}`;
+  toastNotifications.add(`Deleting ${messageId}`);
+
+  for(const calendar of calendarsToDelete) {
+    const calendarId = calendar.calendar_id;
+    try {
+      await ml.deleteCalendar({ calendarId });
+    } catch (error) {
+      console.log('Error deleting calendar:', error);
+      let errorMessage = `An error occurred deleting calendar ${calendar.calendar_id}`;
+      if (error.message) {
+        errorMessage += ` : ${error.message}`;
+      }
+      toastNotifications.addDanger(errorMessage);
```

Comment:
if lots of deletes fail, this will display a lot of errors.
instead of showing a toast per failure, an error list could be built up and displayed in a single toast at the end.
Ideally, a multi-delete action should be behind one endpoint to make it more atomic. 
a `deleteCalendars` endpoint could be added which would take a list of ids.
probably for a follow up PR if you agree it's needed.

Reply:
Good point, @jgowdyelastic. I kept the multiple toasts since we're using the same logic in `Filter Lists` and wanted to keep it consistent.

Not sure if changing to a single toast would make it any easier to read if there were a lot of events listed in one toast - seems like that might be difficult to read as well. 🤔Happy to hear more thoughts on this.

As far as adding a multi-delete endpoint - I agree with that idea. Sounds like a good candidate for a follow-up PR. 😄

Reply:
yeah, just checked the jobs list, and every failed deletion is given it's own toast, so ignore my comment.
I think this is something we should review and address in the future. If more than a few actions fail, do we even need to list them all. Could a `17 jobs failed to delete` message be enough. Obviously something has going wrong pretty badly.
![image](https://user-images.githubusercontent.com/22172091/49724602-98c6e900-fc61-11e8-9e9d-dd36c6c358c5.png)